### PR TITLE
conditions were being duplicated since we broke vs. continuing

### PR DESCRIPTION
**Pull request by dprotaso**

### CODE REVIEW

File: apis/duck/v1/status_types.go [hunk 1]
```diff
@@ -107,19 +107,21 @@ func (source *Status) ConvertTo(ctx context.Context, sink *Status, predicates ..
...
 			conditions = append(conditions, c)
-			break
+			continue
 		}
 
 		for _, predicate := range predicates {
 			if predicate(c.Type) {
 				conditions = append(conditions, c)
-				break
+				continue condition_loop
```
Comment (vagababov) [inline apis/duck/v1/status_types.go]: this I think was correct.

File: apis/duck/v1/status_types.go [hunk 2]
```diff
@@ -107,19 +107,21 @@ func (source *Status) ConvertTo(ctx context.Context, sink *Status, predicates ..
 	sink.ObservedGeneration = source.ObservedGeneration
 
 	conditions := make(apis.Conditions, 0, len(source.Conditions))
+
+condition_loop:
 	for _, c := range source.Conditions {
 		switch c.Type {
 		// Copy over the "happy" condition, which is the only condition that
```
Comment (vagababov) [inline apis/duck/v1/status_types.go]: it might be better constructed as
```
if a == b || a == c {
  ... 
  continue
}
```

Comment (dprotaso) [reply apis/duck/v1/status_types.go]: agreed

**Pull request merged**